src/tdsops.f90

```diff
@@ -88,9 +89,23 @@ function tdsops_init(tds_n, delta, operation, scheme, bc_start, bc_end, &
     logical, optional, intent(in) :: sym !! (==npaire), only for Neumann BCs
     real(dp), optional, intent(in) :: c_nu, nu0_nu !! params for hypervisc.
 
-    integer :: n_stencil
+    integer :: n, n_stencil
+
+    tdsops%n_tds = n_tds
 
-    tdsops%tds_n = tds_n
+    ! we need special treatment in the right-hand-side build stage for
+    ! the very last point in the domain if output length is smaller than
+    ! the input length
+    if (present(from_to)) then
+      if ((bc_end == BC_NEUMANN .or. bc_end == BC_DIRICHLET) &
```

Comment:
Does it not also apply in the case of `bc_start`?

Reply:
Nope, its only at the end. from v to p we have one less data point and we always deal with it at the very end.